src/cborinternal_p.h

```diff
@@ -37,7 +44,25 @@
 #endif
 
 #ifndef CBOR_NO_HALF_FLOAT_TYPE
-#  if defined(__F16C__) || defined(__AVX2__)
+/* Check for FLT16_MANT_DIG using integer comparison. Clang headers incorrectly
+ * define this macro unconditionally when __STDC_WANT_IEC_60559_TYPES_EXT__
+ * is defined (regardless of actual support for _Float16).
+ */
+#  if FLT16_MANT_DIG > 0 || __FLT16_MANT_DIG__ > 0
+#    include <string.h>
```

Comment:
Can you move this `#include` to the top? It won't hurt there.

Reply:
Done